packages/core/src/domain/tracekit/tracekit.spec.ts

```diff
@@ -1,165 +1,110 @@
 import { disableJasmineUncaughtErrorHandler } from '../../../test/specHelper'
-import { subscribe, unsubscribe, traceKitWindowOnError } from './tracekit'
-import { Handler, StackFrame } from './types'
+import { startUnhandledErrorCollection } from './tracekit'
+import { Callback } from './types'
 
-describe('traceKitWindowOnError', () => {
+describe('startUnhandledErrorCollection', () => {
   const testLineNo = 1337
 
-  let subscriptionHandler: Handler | undefined
+  let callbackSpy: jasmine.Spy<Callback>
+  let stopCollectingUnhandledError: () => void
   let resetJasmineUncaughtErrorHandler: () => void
 
   beforeEach(() => {
     ;({ reset: resetJasmineUncaughtErrorHandler } = disableJasmineUncaughtErrorHandler())
+    callbackSpy = jasmine.createSpy()
+    ;({ stop: stopCollectingUnhandledError } = startUnhandledErrorCollection(callbackSpy))
   })
 
   afterEach(() => {
+    stopCollectingUnhandledError()
     resetJasmineUncaughtErrorHandler()
   })
 
   describe('with undefined arguments', () => {
-    it('should pass undefined:undefined', (done) => {
+    it('should pass undefined:undefined', () => {
       // this is probably not good behavior;  just writing this test to verify
       // that it doesn't change unintentionally
-      subscriptionHandler = (stack) => {
-        expect(stack.name).toBeUndefined()
-        expect(stack.message).toBeUndefined()
-        unsubscribe(subscriptionHandler!)
-        done()
-      }
-      subscribe(subscriptionHandler)
-      traceKitWindowOnError(undefined!, undefined, testLineNo)
+      window.onerror!(undefined!, undefined, testLineNo)
+      const [stack] = callbackSpy.calls.mostRecent().args
+      expect(stack.name).toBeUndefined()
+      expect(stack.message).toBeUndefined()
     })
   })
 
   describe('when no 5th argument (error object)', () => {
-    it('should separate name, message for default error types (e.g. ReferenceError)', (done) => {
-      subscriptionHandler = (stack) => {
-        expect(stack.name).toEqual('ReferenceError')
-        expect(stack.message).toEqual('foo is undefined')
-        unsubscribe(subscriptionHandler!)
-        done()
-      }
-      subscribe(subscriptionHandler)
-      traceKitWindowOnError('ReferenceError: foo is undefined', 'http://example.com', testLineNo)
+    it('should separate name, message for default error types (e.g. ReferenceError)', () => {
+      window.onerror!('ReferenceError: foo is undefined', 'http://example.com', testLineNo)
+      const [stack, ,] = callbackSpy.calls.mostRecent().args
```

Comment:
```suggestion
      const [stack] = callbackSpy.calls.mostRecent().args
```
or do we need it?
(other occurrences with or without)

Reply:
Unneeded, I'll remove those.